### PR TITLE
[Remove Spring extender usages] LPS-159168 

### PR DESCRIPTION
**Pull request by JanisZhang**
Hi @dantewang,

This PR is to remove spring extender usage in **`commerce-currency-service`**, thanks!
https://issues.liferay.com/browse/LPS-159168

### CODE REVIEW

File: modules/apps/commerce/commerce-currency-service/src/main/java/com/liferay/commerce/currency/service/impl/CommerceCurrencyLocalServiceImpl.java [hunk 1]
```diff
@@ -459,6 +437,24 @@ public void updateExchangeRates() throws PortalException {
 			ArrayUtil.toLongArray(commerceCurrencyFinder.getCompanyIds()));
 	}
 
+	@Activate
+	protected void activate(BundleContext bundleContext) {
+		super.setAopProxy(commerceCurrencyLocalService);
```
Comment (dantewang) [inline modules/apps/commerce/commerce-currency-service/src/main/java/com/liferay/commerce/currency/service/impl/CommerceCurrencyLocalServiceImpl.java:442]: @JanisZhang Usually we don't call `super.setAopProxy` in `activate`. Did you add it in order to make the service registration work below? If so, please convert the `PortalInstanceLifecycleListenerImpl` an OSGi component referencing this local service, for example, `com.liferay.commerce.account.internal.model.listener.PortalInstanceLifecycleListenerImpl`.

**Review by dantewang — CHANGES_REQUESTED** [review]
@JanisZhang please see my comment and update the PR, thanks!

File: modules/apps/commerce/commerce-currency-service/src/main/java/com/liferay/commerce/currency/service/impl/CommerceCurrencyLocalServiceImpl.java [hunk 2]
```diff
@@ -64,12 +64,15 @@
...
 import org.osgi.framework.ServiceRegistration;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
 
 /**
  * @author Andrea Di Giorgi
  * @author Marco Leo
  * @author Alessio Antonio Rendina
  */
+@Component(enabled = false, service = AopService.class)
```
Comment (dantewang) [inline modules/apps/commerce/commerce-currency-service/src/main/java/com/liferay/commerce/currency/service/impl/CommerceCurrencyLocalServiceImpl.java:75]: @JanisZhang For now, please add `property = "model.class.name=xxx"` to the component annotation, like other local services e.g. `BlogsEntryLocalServiceImpl`. If we get "no" answer from the Slack thread, then we can remove it.

Comment (JanisZhang) [reply modules/apps/commerce/commerce-currency-service/src/main/java/com/liferay/commerce/currency/service/impl/CommerceCurrencyLocalServiceImpl.java:75]: @dantewang OK

**Comment by JanisZhang** [comment]
Hi  @dantewang , force-pushed based on your comments, thanks!

**Comment by JanisZhang** [comment]
ci:stop

**Comment by JanisZhang** [comment]
ci:test:sf

**Comment by JanisZhang** [comment]
ci:test:relevant

**Comment by JanisZhang** [comment]
ci:test:acceptance

File: modules/apps/commerce/commerce-currency-service/src/main/java/com/liferay/commerce/currency/service/impl/CommerceCurrencyLocalServiceImpl.java [hunk 3]
```diff
@@ -463,6 +441,22 @@ public void updateExchangeRates() throws PortalException {
 			ArrayUtil.toLongArray(commerceCurrencyFinder.getCompanyIds()));
 	}
 
+	@Activate
+	protected void activate(BundleContext bundleContext) {
```
Comment (dantewang) [inline modules/apps/commerce/commerce-currency-service/src/main/java/com/liferay/commerce/currency/service/impl/CommerceCurrencyLocalServiceImpl.java:445]: @JanisZhang Once you converted the `PortalInstanceLifecycleListenerImpl` to an OSGi component, you no longer need this service register and unregister logic. So, this class doesn't need the `activate` and `deactivate` method.

**Comment by JanisZhang** [comment]
ci:stop

**Comment by JanisZhang** [comment]
ci:test:sf

**Comment by JanisZhang** [comment]
ci:test:relevant

**Comment by JanisZhang** [comment]
ci:test:acceptance

**Review by dantewang — APPROVED** [review]

File: modules/apps/commerce/commerce-currency-service/src/main/java/com/liferay/commerce/currency/internal/model/listener/PortalInstanceLifecycleListenerImpl.java [hunk 4]
```diff
@@ -46,6 +54,7 @@ public void portalInstanceRegistered(Company company) throws Exception {
 			_commerceCurrencyLocalService, company);
 	}
 
-	private final CommerceCurrencyLocalService _commerceCurrencyLocalService;
+	@Reference
```
Comment (tinatian) [inline modules/apps/commerce/commerce-currency-service/src/main/java/com/liferay/commerce/currency/internal/model/listener/PortalInstanceLifecycleListenerImpl.java:57]: @JanisZhang You should do this commit together with the removal of registration logic inside CommerceCurrencyLocalServiceImpl, this way, people will get what you are doing directly. It confuses me why you are removing the registration logic until I found the conversation between you and Dante.

**Comment by tinatian** [comment]
@JanisZhang Please see my comment and resend.


Tina

**Pull request closed without merging**